doc/cascadia/profiles.schema.json

```diff
@@ -4,7 +4,7 @@
   "title": "Microsoft's Windows Terminal Settings Profile Schema",
   "definitions": {
     "KeyChordSegment": {
-      "pattern": "^(?<modifier>(ctrl|alt|shift)(?:\\+(ctrl|alt|shift)(?<!\\2))?(?:\\+(ctrl|alt|shift)(?<!\\2|\\3))?\\+)?(?<key>[^\\s+]|app|menu|backspace|tab|enter|esc|escape|space|pgup|pageup|pgdn|pagedown|end|home|left|up|right|down|insert|delete|(?<!shift.+)(?:numpad_?[0-9]|numpad_(?:period|decimal))|numpad_(?:multiply|plus|add|minus|subtract|divide)|f[1-9]|f1[0-9]|f2[0-4]|plus)$",
+      "pattern": "^(?<modifier>(ctrl|alt|shift|win)(?:\\+(ctrl|alt|shift|win)(?<!\\2))?(?:\\+(ctrl|alt|shift|win)(?<!\\2|\\3))?\\+)?(?:\\+(ctrl|alt|shift|win)(?<!\\2))?(?<key>[^\\s+]|app|menu|backspace|tab|enter|esc|escape|space|pgup|pageup|pgdn|pagedown|end|home|left|up|right|down|insert|delete|(?<!shift.+)(?:numpad_?[0-9]|numpad_(?:period|decimal))|numpad_(?:multiply|plus|add|minus|subtract|divide)|f[1-9]|f1[0-9]|f2[0-4]|plus)$",
```

Comment:
I think the negative lookbehind backreference part right at the end, which says `!\\2`, needs to say `!\\2|\\3|\\4` .. otherwise, `ctrl+alt+shift+ctrl` becomes valid (since we're only checking the first position for a duplicate)

Reply:
ig ot that wrong. This would DISALLOW only `ctrl+alt+shift+ctrl` but it would ALLOW `ctrl+alt+shift+alt` because it's only checking the first position

Reply:
Okay so clearly, I didn't check the schema, cause this looks like [brainfuck](https://en.wikipedia.org/wiki/Brainfuck) to me. does the regex actually work to disallow `ctrl+alt+ctrl` currently? What kind of regex is this so I can slap it in a web regex tester and play with it?

Reply:
@zadjii-msft... it's "perl compatible regular expressions"

Reply:
Or nevermind it isn't because of the infinite repetition on the look behind...sorry... this is weird.

Reply:
I got it to work under RegexBuddy with C#/.NET config and found 3 things:
1. What Dustin said is right. The new mod capture group needs to have the negative backreference to the previous 3 captures
2. The new 4th modifier capture group was placed outside the `?<modifier>` group so it doesn't actually work right. It needs to go back inside immediately after the `))?` of the 3rd modifier group
3. RegexBuddy got really mad at me for mixing named and numbering capture groups. We could make this maybe slightly more human readable by doing `?<mod1>` and `?<mod2>` and whatnot in each modifier subgroup and then doing `\k<mod1>|\k<mod2> `and so on in the negative backreferences over `\2|\3|\4`...

Reply:
This is what I wrote that worked in RegexBuddy. We'd have to test it in Terminal as obviously we have some weird regex dialect going on here.

`^(?<modifier>(?<mod1>ctrl|alt|shift|win)(?:\+(?<mod2>ctrl|alt|shift|win)(?<!\k<mod1>))?(?:\+(?<mod3>ctrl|alt|shift|win)(?<!\k<mod1>|\k<mod2>))?(?:\+(?<mod4>ctrl|alt|shift|win)(?<!\k<mod1>|\k<mod2>|\k<mod3>))?\+)?(?<key>[^\s+]|app|menu|backspace|tab|enter|esc|escape|space|pgup|pageup|pgdn|pagedown|end|home|left|up|right|down|insert|delete|(?<!shift.+)(?:numpad_?[0-9]|numpad_(?:period|decimal))|numpad_(?:multiply|plus|add|minus|subtract|divide)|f[1-9]|f1[0-9]|f2[0-4]|plus)$`